local/compose/build.go

```diff
@@ -137,7 +137,33 @@ func (s *composeService) ensureImagesExists(ctx context.Context, project *types.
 			utils.DisplayScanSuggestMsg()
 		}
 	}
-	return err
+
+	return s.updateServiceImages(ctx, project)
+}
+
+func (s *composeService) updateServiceImages(ctx context.Context, project *types.Project) error {
+	imageNames := []string{}
+	for _, s := range project.Services {
+		imgName := s.Image
+		if imgName == "" {
+			imgName = getImageName(s, project.Name)
+		}
+		if !utils.StringContains(imageNames, imgName) {
+			imageNames = append(imageNames, imgName)
+		}
+	}
+	images, err := s.getImages(ctx, imageNames)
```

Comment:
you don't need this
on build we ignore buildx response (line 197 `<response>, err = build.Build(...)`)
same as https://github.com/docker/app/blob/9d2c67f87b7338eb1a0fa2f18eb81af3d2aac0e1/internal/commands/build/build.go#L340 you can get the image digest and update the service configuration accordingly, no need to inspect image post-build.